foreman/data_refinery_foreman/surveyor/harmony.py

```diff
@@ -0,0 +1,699 @@
+                                      'RNA was extracted from frozen material using the '
+                                      'RNeasy Fibrous Tissue kit (Qiagen), which included '
+                                      'proteinase K treatment (10 min at 55˚C) of the '
+                                      'lysed sample in RLT-buffer and on-column DNase '
+                                      'treatment. RNA was quantified using a Nanodrop '
+                                      '(NanoDrop technologies) and evaluated for '
+                                      'degradation with a RNA 6000 Nano Labchip on the '
+                                      '2100 Bioanalyzer (Agilent Technologies)'],
+             'geo_accession': ['GSM808778'],
+             'hyb_protocol': ['The standard Illumina hybridization protocol was used. In '
+                              'brief, the samples were hybridized to the arrays at 58ºC '
+                              'overnight.'],
+             'label_ch1': ['biotin'],
+             'label_protocol_ch1': ['100 ng of total RNA was converted to cDNA and '
+                                    'subsequently labeled cRNA using the Ambion Illumina '
+                                    'TotalPrep RNA Amplification kit (Ambion) according to '
+                                    'manufacturer’s instructions'],
+             'last_update_date': ['Feb 06 2013'],
+             'molecule_ch1': ['total RNA'],
+             'organism_ch1': ['Homo sapiens'],
+             'platform_id': ['GPL6102'],
+             'scan_protocol': ['The beadChips were scanned using the Illumina BeadArray '
+                               'Reader, using the standard Illumina scanning protocol'],
+             'series_id': ['GSE32628', 'GSE32969', 'GSE32979'],
+             'source_name_ch1': ['cutaneous squamous cell carcinoma'],
+             'status': ['Public on Feb 06 2013'],
+             'submission_date': ['Oct 05 2011'],
+             'supplementary_file': ['NONE'],
+             'taxid_ch1': ['9606'],
+             'title': ['SCC_P-39'],
+             'type': ['RNA']}
+
+    """
+
+    # Prepare the harmonized samples
+    original_samples = metadata.copy()
+    harmonized_samples = {}
+
+    ##
+    # Title!
+    # We also use the title as the key in the returned dictionary
+    ##
+    for sample in original_samples:
+        title = extract_title(sample)
+        # If we can't even find a unique title for this sample
+        # something has gone horribly wrong.
+        if not title:
+            return {}
+        sample['title'] = title
+        harmonized_samples[title] = {}
+
+    ##
+    # Sex!
+    ##
+    sex_fields = [    
+                    'sex',
+                    'gender',
+                    'subject gender',
+                    'subjext sex',
+                    'characteristic [sex]',
+                    'characteristics [sex]',
```

Comment:
For sex, age, and part you include the `characteristic [{item}]` in this list, but it looks like `add_variants` will take care of those variants for you.

Reply:
This looks reduntant, but there are some samples actually have a nested characteristic field. Added a comment about it.